_posts/2018-03-01-spatial-modeling-mlr.markdown

```diff
@@ -0,0 +1,318 @@
+---
+author: Patrick Schratz
+biblio-style: apalike
+bibliography: ../bibs/spatial\_modeling\_`mlr`.bib
+categories: r
+comments: True
+date: 01 March, 2018
+layout: post
+---
+link-citations: True
+title: A practical guide to performance estimation of spatially tuned
+  machine-learning models for spatial data using `mlr`
+
+<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
+* TOC 
+{:toc}
+
+\[[view raw
+Rmd](https://raw.githubusercontent.com//r-spatial/r-spatial.org/gh-pages/_rmd/2018-03-01-spatial-modeling-mlr.Rmd)\]
+
+Introduction
+============
+
+Recently we started to integrate the spatial partitioning methods of
+[sperrorest](https://mlr-org.github.io/mlr-tutorial/devel/html/handling_of_spatial_data/index.html)
+into [mlr](http://mlr-org.github.io). `mlr` is a unified interface to
+conduct all kind of modeling (similar to `caret`), currently supporting
+more than 100 modeling packages. In comparison to sperrorest it
+excelaertes by providing the option to easily tune hyperparameters. We
+started with the most common spatial partitioning approach which uses
+k-means clustering (see Brenning ([2012](#ref-sperrorest))). In this
+blog post I will give a practical guide on how to perform a nested
+spatial cross-validation in `mlr`. I will use a SVM as the algorithm
+because it is a widely know and used algorithm that always needs to be
+tuned to achieve good performance. Although the tuning effect of models
+with meaningful default hyperparameter values such as Random Forest is
+not high, one should always conduct a hyperparameter tuning as the
+specific effect is not known a priori.
+
+Everything in `mlr` is build upon the following steps: Create a task,
+(tune hyperparameters), train your model, predict to a new data set.
+When the model performance is to be estimated, this procedure is done
+hundreds of times to reduce the variance among different traing +
+predict arangements. It essentially mimics the situation that you would
+apply your fitted model on 500 different unkown test data sets. You will
+get a different performance on each. However, usually you have no
+in-situ data to check whether your model
+
+Creating a task
+===============
+
+The task essentially stores all information around your data set: Type
+of the response variable, supervised/unsupervised, spatial or
+non-spatial and many more.
+
+I will use the example data set for spatial applications `ecuador` that
+was added to `mlr`. This data set from Jannes Muenchow also comes with
+sperrorest. Since the integration is quite new, we need to use the
```

Comment:
Why have the same data in two packages? Is it going to be removed from one of the two?

Reply:
`mlr` creates a `Task` object from the data set. A task consists of more than just the data, e.g. metadata information.
So essentially you don't call `data("ecuador")` in `mlr` to get a simple data.frame but `data("spatial.task")` which gives you a `Task` object. 

This is needed for internal `mlr` tests to avoid the need of recreating the task from scratch every time.